ui/views.py

```diff
@@ -154,6 +157,32 @@ def get_context_data(self, video, **kwargs):  # pylint: disable=arguments-differ
         return context
 
 
+class VideoDownload(VideoDetail):
+    """
+    Download a public video
+    """
+    def download(self, video):
+        """
+        Return the response as a file download
+        Returns:
+            HttpResponse: the response videofile content and a file attachment header
```

Comment:
Looks like this should be a `StreamingHttpResponse`

Reply:
✅